Framework/CurveFitting/src/Algorithms/QENSFitSimultaneous.cpp

```diff
@@ -164,10 +165,24 @@ WorkspaceGroup_sptr makeGroup(Workspace_sptr workspace) {
   return group;
 }
 
+IAlgorithm_sptr createProcessIndirectFitParametersAlgorithm(
+    ITableWorkspace_sptr parameterWorkspace, const std::string &xAxisUnit,
+    const std::vector<std::string> &parameterNames) {
+  auto pifp =
+      AlgorithmManager::Instance().create("ProcessIndirectFitParameters");
+  pifp->setAlwaysStoreInADS(false);
+  pifp->setProperty("InputWorkspace", parameterWorkspace);
+  pifp->setProperty("ColumnX", "axis-1");
+  pifp->setProperty("XAxisUnit", xAxisUnit);
+  pifp->setProperty("ParameterNames", parameterNames);
+  return pifp;
+}
+
 ITableWorkspace_sptr transposeFitTable(ITableWorkspace_sptr table,
-                                       IFunction_sptr function) {
+                                       IFunction_sptr function,
+                                       const std::string &yAxisType) {
```

Comment:
In  the transposeFitTable method, the function parameter never needs to be a shared_ptr, cannot be null and should not be changed. It should preferably be passed by const reference.